content/news/XXXX-XX-XX-dvs-internals-pt3.md

```diff
@@ -0,0 +1,219 @@
+
+A simple discrete derivative operation achieves this:
+
+$$
+y[n] = x[n] - x[n-1] \tag{1}
+$$
+
+$\text{where:}$<br>
+$\text{- x[n]: Input sample}$<br>
+$\text{- x[n-1]: Delayed input sample}$<br>
+$\text{- y[n]: Difference of both values}$<br>
+<br>
+
+When applied to the offset-modulated signal, we get:
+
+![Traktor MK2 signal with derivative]({static}/images/news/traktor-mk2-signal-with-derivative.svg)
+
+The resulting waveform oscillates cleanly around zero, which is ideal for
+analysis. It also makes it easier to pinpoint the half-cycle peaks needed for
+bit detection.
+
+---
+
+## Demodulation Techniques
+
+To extract bits from the signal, we detect the zero-crossings and sample the
+amplitude of the sinusoid at those moments.
+
+![Traktor MK2 signal with
+zero-crossings]({static}/images/news/traktor-mk2-signal-with-zero-crossings.svg)
+
+You may notice that the derivative’s zero-crossings don’t align perfectly with
+the original peaks. That’s due to a delay introduced by the filter. Smoothing
+the signal first, then compensating for the delay (e.g., by selecting
+$x[n-3]$), yields better results.
+
+For greater accuracy, one could analyze the phase response $\phi(\omega)$,
+which shows how filter delay varies with input frequency - but for this use case,
+a fixed delay works well enough.
+
+The filtered signal can cross the x-axis in two directions-positive to
+negative or vice versa. Based on the direction, we determine which half-cycle
+contains the encoded bit.
+
+![Traktor MK2 signal with readings]({static}/images/news/traktor-mk2-signal-with-readings.svg)
+
+Demodulation is then as simple as applying a threshold: amplitudes above it
+are **1**, and below it are **0**.
+
+![Traktor MK2 signal with timecodes]({static}/images/news/traktor-mk2-signal-with-timecodes.svg)
+
+On actual vinyl, the physical behavior of the needle causes the offset to
+decay over time, because the needle slowly drifts back to the middle. This decay complicates bit extraction.
+
+To compensate, we analyze the **slope** between subsequent readings by
+reusing the derivative equation in $\text{(1)}$.
+
+$$slope[n] = reading[n] - reading[n-1]$$
+
+$\text{where:}$<br>
+$\text{- x[n]: Current reading}$<br>
+$\text{- x[n-1]: Last reading}$<br>
+$\text{- y[n]: Difference of both values}$<br>
+<br>
+
+![Traktor MK2 signal with indicated slope]({static}/images/news/traktor-mk2-signal-with-slope.png)
+
+This method helps isolate the encoded signal from the floating zero line
+caused by mechanical drift.
+
+---
+
+## The Code
+
+> **Note:** A deeper explanation of LFSRs can be found in [DVS Internals Pt. 2]({filename}/news/2021-12-22-dvs-internals-pt2.md)
+
+Interestingly, the Traktor MK2 system also uses a [Linear Feedback Shift Register](https://en.wikipedia.org/wiki/Linear-feedback_shift_register) - but with different properties. While Serato’s LFSR has a
+**20-bit** length, Traktor’s appears to use a **110-bit** register with a minimum
+run length of two symbols.
```

Comment:
how did you come to that 110-bit conclusion? Berlekamp–Massey algorithm?

Reply:
Would be nice to add as an additional sentence. That makes it easier to follow along because numbers are clearly derived, not just "god given".